Auth0/CredentialsManager.swift

```diff
@@ -362,7 +362,12 @@ public struct CredentialsManager {
 
     private func retrieveCredentials() -> Credentials? {
         guard let data = self.storage.getEntry(forKey: self.storeKey) else { return nil }
-        return try? NSKeyedUnarchiver.unarchivedObject(ofClass: Credentials.self, from: data)
+        do {
+            return try JSONDecoder().decode(Credentials.self, from: data)
+        } catch {
+            print(error.localizedDescription)
```

Comment:
@Vaidios I know that this error was already not being surfaced here, so this is not a comment on your PR but more this process in general. If an `NSCoder` failure is the cause of #862 there is an argument to be made that the problem would have been easier to debug had this function either throw here or have the `.noCredentials` error provide a more descriptive message. What are your thoughts @desusai7 ?

Reply:
I totally agree, it would be much easier to find a root cause if that error would be propagated.
Maybe we should throw an error such as 'credentialsParsingFailure' instead of non-descriptive 'noCredentials'.

PS. For the time being, I will remove that do-catch with print to be optional try, just as before